src/lib/stitching/exchange/v2/stitching.ts

```diff
@@ -546,6 +554,76 @@ export const exchangeStitchingEnvironment = ({
             return offerResult
           },
         },
+        submitOrderOffer: {
+          resolve: async (_source, args, context, info) => {
+            const {
+              artworkId,
+              note,
+              confirmedSetupIntentId,
+              offerId,
+            } = args.input
+            const { submitArtworkInquiryRequestLoader } = context
+
+            const submitOrderWithOffer = await info.mergeInfo.delegateToSchema({
+              schema: exchangeSchema,
+              operation: "mutation",
+              fieldName: "commerceSubmitOrderWithOffer",
+              args: {
+                input: {
+                  offerId,
+                  confirmedSetupIntentId,
+                },
+              },
+              context,
+              info,
+              transforms: [
+                // add orderOrError.order.internalID to the Order selectionSet
+                new WrapQuery(
+                  ["commerceSubmitOrderWithOffer", "orderOrError", "order"],
+                  (selectionSet: SelectionSetNode) => {
+                    const newSelections = [
+                      ...selectionSet.selections,
+                      {
+                        kind: Kind.FIELD,
+                        name: {
+                          kind: Kind.NAME,
+                          value: "internalID",
+                        },
+                      },
+                    ]
+                    return { ...selectionSet, selections: newSelections }
+                  },
+                  (result) => {
+                    return result
+                  }
+                ),
+              ],
+            })
+
+            const { orderOrError } = submitOrderWithOffer
+
+            if (orderOrError.error || !orderOrError.order) {
+              return submitOrderWithOffer
+            }
+
+            try {
+              const {
+                order: { internalID: orderId },
+              } = orderOrError
+              await submitArtworkInquiryRequestLoader({
```

Comment:
@erikdstock I didn't refactor the previous mutation, but maybe we could reuse this loader in the previous mutation, idk the details of how we use it but it looks like they have something in common
Could be a good thing to keep Gravity as the source of truth in terms of linking the conversation with the order there